### ISSUE TITLE
java.lang.RuntimeException: Error receiving broadcast Intent { act=com.google.android.gms.auth.api.phone.SMS_RETRIEVED flg=0x200010 pkg=com.example.sms_flutter (has extras) } in fman.ge.smart_auth.SmartAuthPlugin$SmsBroadcastReceiver@79393a2

### ISSUE THREAD

**Issue by CometNet** [Tkko/flutter_smart_auth#2]
As long as you use getSmsCode to receive the verification code, it starts to flash back, prompting this error

`D[/AndroidRuntime]()(31911): Shutting down VM
E[/AndroidRuntime]()(31911): FATAL EXCEPTION: main
E[/AndroidRuntime]()(31911): Process: com.example.sms_flutter, PID: 31911
E[/AndroidRuntime]()(31911): java.lang.RuntimeException: Error receiving broadcast Intent { act=com.google.android.gms.auth.api.phone.SMS_RETRIEVED flg=0x200010 pkg=com.example.sms_flutter (has extras) } in fman.ge.smart_auth.SmartAuthPlugin$SmsBroadcastReceiver@79393a2
E[/AndroidRuntime]()(31911): 	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1581)
E[/AndroidRuntime]()(31911): 	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
E[/AndroidRuntime]()(31911): 	at android.os.Handler.handleCallback(Handler.java:938)
E[/AndroidRuntime]()(31911): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E[/AndroidRuntime]()(31911): 	at android.os.Looper.loop(Looper.java:233)
E[/AndroidRuntime]()(31911): 	at android.app.ActivityThread.main(ActivityThread.java:8030)
E[/AndroidRuntime]()(31911): 	at java.lang.reflect.Method.invoke(Native Method)
E[/AndroidRuntime]()(31911): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
E[/AndroidRuntime]()(31911): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
E[/AndroidRuntime]()(31911): Caused by: java.lang.NullPointerException: null cannot be cast to non-null type kotlin.String
E[/AndroidRuntime]()(31911): 	at fman.ge.smart_auth.SmartAuthPlugin$SmsBroadcastReceiver.onReceive(SmartAuthPlugin.kt:433)
E[/AndroidRuntime]()(31911): 	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1566)
E[/AndroidRuntime]()(31911): 	... 8 more
I[/Process]() (31911): Sending signal. PID: 31911 SIG: 9`

**Comment by Tkko** [comment]
Resolved in version [1.0.7](https://pub.dev/packages/smart_auth/changelog)

**Comment by PoriyaBytez** [comment]
java.lang.RuntimeException: Error receiving broadcast Intent { act=com.google.android.gms.auth.api.phone.SMS_RETRIEVED flg=0x280010 pkg=(has extras) } in com.google.android.gms.internal.firebase-auth-api.zzafr@147c057

It seems like you're encountering an intermittent error in versions 1.1.1 and 2.0.0. To resolve it, I'd recommend thoroughly checking your code for any potential issues related to state management or resource handling.
I am using kotlin version 1.7.10 and flutter SDK 3.13.5

**Comment by burekas7** [comment]
This is the solution:
```
Pinput(
      smsRetriever: Platform.isAndroid ? smsRetrieverImpl : null,
```

**Comment by Tkko** [comment]
This issue should be fixed in version ^2.0.0
